io/aligned_buf_pool.cc

```diff
@@ -0,0 +1,99 @@
+#include "io/aligned_buf_pool.h"
+
+#include "base/logging.h"
+#include "io/io.h"
+#include "util/fibers/proactor_base.h"
+#include "util/fibers/uring_proactor.h"
+
+namespace io {
+
+namespace {
+
+thread_local AlignedBufPool tl_buf_pool;
+
+}  // namespace
+
+void AlignedBufPool::Init(size_t backing_size) {
+  auto& pool = tl_buf_pool;
+  CHECK_EQ(pool.buffer_, nullptr);
+
+  pool.buffer_size_ = backing_size;
+  pool.buffer_ = new (std::align_val_t(kAlign)) uint8_t[backing_size];
+
+  CHECK_EQ(util::fb2::ProactorBase::me()->GetKind(), util::fb2::ProactorBase::IOURING);
+  auto* uring = static_cast<util::fb2::UringProactor*>(util::fb2::ProactorBase::me());
+
+  iovec vec{static_cast<void*>(pool.buffer_), backing_size};
+  CHECK_EQ(uring->RegisterBuffers(&vec, 1), 0);
+}
+
+void AlignedBufPool::Destroy() {
+  auto& pool = tl_buf_pool;
+  if (!pool.buffer_)
+    return;
+
+  ::operator delete[](pool.buffer_, std::align_val_t(kAlign));
+  pool.buffer_ = nullptr;
+
+  auto* uring = static_cast<util::fb2::UringProactor*>(util::fb2::ProactorBase::me());
+  uring->UnregisterBuffers();
+}
+
+AlignedBuf AlignedBufPool::RequestChunk(size_t length) {
+  std::optional<size_t> offset = tl_buf_pool.RequestSegment((length + kAlign - 1) / kAlign);
+
+  uint8_t* ptr = nullptr;
+  if (offset)
+    ptr = &tl_buf_pool.buffer_[*offset * kAlign];
+  else
+    ptr = new (std::align_val_t(kAlign)) uint8_t[length];
```

Comment:
following the suggestion above, I would also remove the fallback code. 
This implementation enforces no-backpressure policy. I think it's not the right place to commit to this.